src/viur/core/modules/file.py

```diff
@@ -708,45 +707,93 @@ def write(
+            rootnode = self.ensureOwnModuleRootNode()
+        elif not foldername:
+            # if rootnode is set and foldername is not, save the file in the root of the rootnode
+            foldername = []
+
+        if foldername is not None:
+            foldernames = foldername
+            if isinstance(foldername, str):
+                foldernames = foldernames.replace('\\', '/')
+                foldernames = (i for i in foldernames.split('/'))
+            foldernames = (i for i in foldernames if i)
+
+            parentrepokey = rootnode.key
+            parentfolderkey = rootnode.key
+
+            for fname in foldernames:
+                currentfolder = (self.addSkel("node").all()
+                                 .filter("parentrepo", parentrepokey)
+                                 .filter("parententry", parentfolderkey)
+                                 .filter("name", fname)
+                                 .getSkel())
+                if currentfolder:
+                    parentfolderkey = currentfolder.key
+                else:
+                    newfolderskel = self.addSkel("node")
+                    newfolderskel["name"] = fname
+                    newfolderskel["parentrepo"] = parentrepokey
+                    newfolderskel["parententry"] = parentfolderkey
+                    newfolderskel.write()
+                    parentfolderkey = newfolderskel["key"]
+
+        # Save the file into the folder.
         dl_key = utils.string.random()
 
         if public:
             dl_key += PUBLIC_DLKEY_SUFFIX  # mark file as public
 
-        bucket = File.get_bucket(dl_key)
+        bucket = self.get_bucket(dl_key)
 
         blob = bucket.blob(f"{dl_key}/source/{filename}")
         blob.upload_from_file(io.BytesIO(content), content_type=mimetype)
 
-        skel = self.addSkel("leaf")
-        skel["name"] = filename
-        skel["size"] = blob.size
-        skel["mimetype"] = mimetype
-        skel["dlkey"] = dl_key
-        skel["weak"] = True
-        skel["public"] = public
-        skel["width"] = width
-        skel["height"] = height
-        skel["crc32c_checksum"] = base64.b64decode(blob.crc32c).hex()
-        skel["md5_checksum"] = base64.b64decode(blob.md5_hash).hex()
-
-        skel.write()
-        return skel["key"]
+        fileskel = self.addSkel("leaf")
+        fileskel["name"] = filename
+        fileskel["size"] = blob.size
+        fileskel["mimetype"] = mimetype
+        fileskel["dlkey"] = dl_key
+        fileskel["weak"] = foldername is None
+        fileskel["public"] = public
+        fileskel["width"] = width
+        fileskel["height"] = height
+        fileskel["crc32c_checksum"] = base64.b64decode(blob.crc32c).hex()
+        fileskel["md5_checksum"] = base64.b64decode(blob.md5_hash).hex()
+
+        if foldername is not None:
+            fileskel["parentrepo"] = parentrepokey
+            fileskel["parententry"] = parentfolderkey
+            fileskel["pending"] = False
```

Comment:
Shouldn't `fileskel["pending"] = False` always be the case here in write? The "upload" is done at once.

```suggestion
```